### ISSUE TITLE
Keptn CLI OAuth Login silently fails in case I have something running on port 3000 already

### ISSUE THREAD

**Issue by christian-kreuzberger-dtx** [keptn/keptn#8102]
With Keptn CLI 0.15.1, I tried

```
$ keptn auth --oauth ...
```

This silently failed, as on port 3000 I have another application running already on my system.

**Comment by thisthat** [comment]
Hey @christian-kreuzberger-dtx, I would consider this an enhancement and not a bug report. 
A few implementation details: OAuth requires validation of the redirect URL and the port must be part of such URL. Hence, the CLI cannot simply look for an open port but must be passed as a parameter based on the configuration of the user OAuth application.

**Comment by christian-kreuzberger-dtx** [comment]
Enhancement/improvement works for me :+1:

Some more thoughts from my side:

1. If it's an oauth implementation detail, it should be documented. I looked through the docs here https://keptn.sh/docs/0.16.x/operate/user_management/ and never did it mention port 3000 at all.
2. If port 3000 is strong a requirement, I would suggest implementing a simple check in the CLI and print a human readable error message containing a link to the docs.
3. To me this smells like something that should be configurable. I've never seen port 3000 being a requirement for oauth at all. I understand that the oauth-server needs to be configured with a callback address, but surely something like `localhost:*` or `localhost:[3000-3099]` should be possible

**Comment by DarthBenro008** [comment]
I agree with @christian-kreuzberger-dtx, port `3000` is very widely used by developers and many applications (eg: grafana on local docker). 

As its an oauth, the scope of the client listening to the port is very less (it will close once it receives a redirect).

I highly suggest to migrate the operation from `3000` to any another [IANA Unassigned Port](https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=unassigned&page=2)

I personally suggest port number: `4507` (simply because its a primenumber which is IANA unassigned)

I'd be happy to make a PR regarding the same! 😃

**Comment by MdSahil-oss** [comment]
@thisthat I would lke to work on this :)

**Comment by thisthat** [comment]
Thank you @MdSahil-oss :)

**Comment by MdSahil-oss** [comment]
Hi, @thisthat I'm thinking to add `oauth-port` new flag into CLI to take user defined port number as input and start oauth authentication on that port number if port `3000` is busy otherwise oauth authentication will be done by port `3000` by default.

Let me know what do you think of this.
